sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -73,32 +75,48 @@ protected ServiceBusMessage fromByte(byte[] payload) {
+        getStringHeader(headers, copyHeaders, MessageHeaders.REPLY_CHANNEL).ifPresent(message::setReplyTo);
 
         // AzureHeaders.
-        getStringHeader(headers, AzureHeaders.RAW_ID).ifPresent(message::setMessageId);
-        Optional.ofNullable(headers.get(AzureHeaders.SCHEDULED_ENQUEUE_MESSAGE, Integer.class))
+        getStringHeader(headers, copyHeaders, AzureHeaders.RAW_ID).ifPresent(message::setMessageId);
+        Optional.of(AzureHeaders.SCHEDULED_ENQUEUE_MESSAGE)
+                .filter(copyHeaders::remove)
+                .map(key -> headers.get(key, Integer.class))
                 .map(Duration::ofMillis)
                 .map(Instant.now()::plus)
                 .map((ins) -> OffsetDateTime.ofInstant(ins, ZoneId.systemDefault()))
                 .ifPresent(message::setScheduledEnqueueTime);
 
         // ServiceBusMessageHeaders, service bus headers have highest priority.
-        getStringHeader(headers, MESSAGE_ID).ifPresent(message::setMessageId);
-        Optional.ofNullable((Duration) headers.get(TIME_TO_LIVE)).ifPresent(message::setTimeToLive);
-        Optional.ofNullable((Instant) headers.get(SCHEDULED_ENQUEUE_TIME))
+        getStringHeader(headers, copyHeaders, MESSAGE_ID).ifPresent(message::setMessageId);
+
+        Optional.of(TIME_TO_LIVE)
+                .filter(copyHeaders::remove)
+                .map(key -> headers.get(key, Duration.class))
+                .ifPresent(message::setTimeToLive);
+
+        Optional.of(SCHEDULED_ENQUEUE_TIME)
+                .filter(copyHeaders::remove)
+                .map(key -> headers.get(key, Instant.class))
                 .map((ins) -> OffsetDateTime.ofInstant(ins, ZoneId.systemDefault()))
                 .ifPresent(message::setScheduledEnqueueTime);
-        getStringHeader(headers, SESSION_ID).ifPresent(message::setSessionId);
-        getStringHeader(headers, CORRELATION_ID).ifPresent(message::setCorrelationId);
-        getStringHeader(headers, TO).ifPresent(message::setTo);
-        getStringHeader(headers, REPLY_TO_SESSION_ID).ifPresent(message::setReplyToSessionId);
-        getStringHeader(headers, PARTITION_KEY).ifPresent(message::setPartitionKey);
 
-        headers.forEach((key, value) -> message.getApplicationProperties().put(key, value.toString()));
+        getStringHeader(headers, copyHeaders, SESSION_ID).ifPresent(message::setSessionId);
+        getStringHeader(headers, copyHeaders, CORRELATION_ID).ifPresent(message::setCorrelationId);
+        getStringHeader(headers, copyHeaders, TO).ifPresent(message::setTo);
+        getStringHeader(headers, copyHeaders, REPLY_TO_SESSION_ID).ifPresent(message::setReplyToSessionId);
+        getStringHeader(headers, copyHeaders, PARTITION_KEY).ifPresent(message::setPartitionKey);
+
+        copyHeaders.forEach(key -> {
+            message.getApplicationProperties().put(key, headers.get(key).toString());
```

Comment:
the return type of this method `com.azure.messaging.servicebus.ServiceBusMessage#getApplicationProperties` is a `Map<String,Object>`, why don't we just pass in the object value type ?

Reply:
To convert values as String is to avoid non built-in types cannot be encoded when sending AMQP messages. This is actually not a good solution but a bug I think, because when convert a non built-in object to String, it cannot be restored. We could mark it and investigate how to resolve it.